### PR TITLE
Issue5685 campaign_language and campaign_country to Message Broker transactional requeues

### LINKED ISSUE #5685: campaign_language in Campaign Signup and Reportback Transactions
When a user signs up (`campaign_signup`) for a campaign or reports back (`campaign_reportback`) a request for a transactional message is sent to the Message Broker system. The payload of the message request needs to include the `campaign_language`. The value comes from the language setting for the related campaign node.

**NOTE**: This is different than the `user_language` and `user_country` which are currently in the request payload but does not address this issue.

**Example payload**:

```
(
    [activity] => campaign_signup
    [email] => mfantini+pttranslator@dosomething.org
    [uid] => 3202400
    [campaign_language] => en-global
    [merge_vars] => Array
        (
            [MEMBER_COUNT] => 4.3 million
            [FNAME] => Brazil
            [CAMPAIGN_TITLE] => Sea Turtle Street Sweep
            [CAMPAIGN_LINK] => https://thor.dosomething.org/us/campaigns/sea-turtle-street-sweep?source=node/3624
            [CALL_TO_ACTION] => Save sea turtles by picking up trash before it reaches the ocean.
            [STEP_ONE] => Clean Up Trash!
            [STEP_TWO] => Snap a Pic
            [STEP_THREE] => Haul It Away
        )

    [user_country] => US
    [user_language] => en
    [email_template] => mb-campaign-signup-US
    [subscribed] => 1
    [event_id] => 3624
    [email_tags] => Array
        (
            [0] => 3624
            [1] => drupal_campaign_signup
        )

    [activity_timestamp] => 1446224947
    [application_id] => US
)
```

**Related Issues**:
- https://github.com/DoSomething/phoenix/issues/5601
- https://github.com/DoSomething/phoenix/issues/5804

### PR DESCRIPTION
**Pull request by deezone**
- #5685 (Fixes Campaign signups only)

Adds `campaign_language` and `campaign_country` to $params sent to Message Broker request for campaign signups.

**Testing**: The payload produced by the `dosomething_mbp.module` before sending a message to Message Broker. Note the `CAMPAIGN_TITLE` and  `email_template` ("Please Don't Break Anything" & "mb-campaign-signup-US") are based on the `campaign _language` ("en") not the `user_language` ("en-global"):

**Example** - Global user signing up for US campaign. Expect campaign language rather than user language to define the language->country for the template name and the language of the `merge_var` content.

```
Array
(
    [activity] => campaign_signup
    [email] => dlee+update-test98@dosomething.org
    [uid] => 1705551
    [merge_vars] => Array
        (
            [MEMBER_COUNT] => 3.5 million
            [FNAME] => Test
            [CAMPAIGN_TITLE] => Please Don't Break Anything
            [CAMPAIGN_LINK] => http://dev.dosomething.org:8888/us/campaigns/global-please-dont-break-anything?source=node/1691
            [CALL_TO_ACTION] => Everything works
            [STEP_ONE] => Do Header
            [STEP_TWO] => Snap a Pic
            [STEP_THREE] => do this after
        )

    [user_country] => IS
    [user_language] => en-global
    [campaign_language] => en
    [campaign_country] => US
    [email_template] => mb-campaign-signup-US
    [subscribed] => 1
    [event_id] => 1691
    [email_tags] => Array
        (
            [0] => 1691
            [1] => drupal_campaign_signup
        )

)
```

### CODE REVIEW

**Comment by deezone** [comment]
@mikefantini This PR as is will get campaign signup transactionals in the language of the campaign. Do you want to release this for testing or wait for the report back solution as well? The report back solution will require additional work.

File: lib/modules/dosomething/dosomething_global/dosomething_global.module [hunk 1]
```diff
@@ -730,3 +730,21 @@ function dosomething_global_user_details($account = NULL) {
...
+ *  - country_code
+ *
+ *  @return array
+ *    Details about the user account:
+ *      - language
+ *      - country_code
+ */
+function dosomething_global_campaign_details($node) {
+
+  $language = $node->language;
```
Comment (deezone) [inline lib/modules/dosomething/dosomething_global/dosomething_global.module]: Crap... this is the base language of the node, not necessarily the language of the submission. Need to check to see what happens when a campaign is in more than language and the user signs up for a campaign in a language other than the default node language. Perhaps there's a global function to lookup the campaign language by submission path which should contain the country code ?

Comment (itsjoekent) [inline lib/modules/dosomething/dosomething_global/dosomething_global.module]: If you want to grab the country code from the URL 

dosomething_global_get_current_prefix() 

However I suggest using dosomething_global_get_language() instead for this

**Comment by mikefantini** [comment]
@DeeZone I defer to @marahml and @namimody on implementation and how they want to test this.

File: lib/modules/dosomething/dosomething_global/dosomething_global.module (hunk 1 above)
Comment (deezone) [inline lib/modules/dosomething/dosomething_global/dosomething_global.module]: Thanks @deadlybutter ,

`dosomething_global_get_current_prefix()` looks good as I'd use the `$respect_url` flag on `dosomething_global_get_language().

**Comment by deezone** [comment]
@sergii-tkachenko Please review... thoughts? I'll make a new PR to get around the conflicts but would still want your feedback and give you some exposure to the proposed changes.

**Comment by sergiitk** [comment]
@DeeZone Is this still relevant?

**Comment by deezone** [comment]
@sergii-tkachenko Yes, still relevant. It needs to be rerolled to allow merging. I'll create a new PR for your review.

**Comment by sergiitk** [comment]
Ok!

**Pull request closed without merging**